python/cudf/cudf/tests/test_concat.py

```diff
@@ -1997,3 +1997,38 @@ def test_concat_dict_incorrect_type_index(d):
         match="cannot concatenate a dictionary containing indices",
     ):
         cudf.concat(d, axis=1)
+
+
+@pytest.mark.parametrize(
+    "axis",
+    [
+        0,
+        pytest.param(
+            1,
+            marks=pytest.mark.xfail(
+                reason=("cannot concatenate indices across axis 1")
+            ),
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "idx",
+    [
+        [cudf.Index([1, 2, 3])],
+        [cudf.Index([1, 2, 3]), cudf.Index([4, 5, 6])],
+        [
+            cudf.MultiIndex(
+                levels=[[1, 2], ["blue", "red"]],
+                codes=[[0, 0, 1, 1], [1, 0, 1, 0]],
+            )
+        ],
+        [cudf.CategoricalIndex([1, 2, 3])],
+    ],
+)
+def test_concat_index(idx, axis):
+    result = cudf.concat(idx, axis=axis)
+    assert isinstance(result, cudf.Index)
```

Comment:
*nit*: in general, prefer asserting for value-based correctness as well as type-correctness. In this case, where I think we actually just want to be ensuring that deprecations are correctly enforced, it's ok not to.

Reply:
sure